SPDY/SPDYSocket.m

```diff
@@ -533,7 +533,6 @@ - (bool)connectToOrigin:(SPDYOrigin *)origin
     }
 
     [self _startConnectTimeout:timeout];
-    _flags |= kDidStartDelegate;
```

Comment:
Great catch. The timeout would fail to notify the delegate, but the PAC response would still complete. This would trigger a connect, effectively voiding the timeout. I've fixed this now and added some unit tests for this case. Thanks.